docs/entities/registeredservice.md

```diff
@@ -0,0 +1,49 @@
+- HealthCheck: A mechanism to be able to verify the service is online and ready to use. One way this can be accomplished is
+by providing an image containing a client that can be run to test connectivity and authentication. This property is optional, when it is absent, it means the service will be considered available as soon as it is registered. 
+- SLA: Provides multiple levels of resiliency, scalability, fault tolerance and security. This allows claims to take into account the robustness of service. This property is optional, when it is absent, it means that there is no distinctions between services given the SLA.
+
+
+## Status
+
+The Status of the Service it is also defined under the [RegisteredService CRD](../../config/crd/bases/primaza.io_registeredservices.yaml). It contains only one required property to track state. The state could be one of the following: registered, available, claimed, unreachable or undetermined. The registered service state is "registered" right after creation. In most cases this will change immediately after the controller gets notified of existence of new resource, unless controller fails even before it tries to run health check. If the health check passes or is not defined, the state will change to "available". Once a RegisteredService is claimed, its state moves to "claimed". On the other hand, if the health check comes back as a failure, the state would change to "unreachable". In some cases, the health check could failed before it can determine where it fails of passes. In those cases, the state of the registered service will go to "undetermined" indicating there is an issue with the health check not with the service itself.
+
+There are some situations when health check will fail while the registered service is in "claimed" state. In those cases the registered service will move to "unreachable". If, at a later time, the health check passes then the controller will check if there is still a claim matching the registered service and move the state back to "claimed". However, if there is not claim matching the registered service the state will move to "available"
+
+## Use Cases
+
+### Creation
+
+When a RegisteredService is created, Primaza will first check if the health check can be run and will modify the state accordingly. The ServiceCatalog will be updated to include the new RegisteredService only if the state can be moved to "available".
+
+### Deletion
+
+When a RegisteredService resource  is deleted, the ServiceCatalog entry for the service should be deleted. Also, an claims for the service will change the state to pending since the matched registered service does not exist any longer. Also, when a RegisteredService resource state changes to "claimed" the corresponding entry in the ServiceCatalog resource will be removed.
```

Comment:
```suggestion
When a RegisteredService resource  is deleted, the ServiceCatalog entry for the service should be deleted. Also, if the service is claimed, the ServiceClaim for the service will change the state to pending since the matched registered service does not exist any longer. Also, when a RegisteredService resource state changes to "claimed" the corresponding entry in the ServiceCatalog resource is removed.
```